src/scheduler/pbs_sched.c

```diff
@@ -689,20 +664,24 @@ read_sched_cmd(int sock)
 		 * and if we get priority command then just ignore it
 		 * since we are not yet in middle of schedule cycle
 		 */
-		cmd_prio = new_sched_cmd();
 		if (cmd_prio != NULL) {
-			int rc_prio = get_sched_cmd_noblk(sock, cmd_prio);
+			int rc_prio = get_sched_cmd_noblk(sock, &cmd_prio);
```

Comment:
NULL check for cmd_prio is removed in the later commit 99593d93461d8a48e3c26de890c4698d6d822bec. 

A single server can at the most send one normal/medium priority command followed by a super high priority command. So maximum Scheduler can get two commands from each server at a time. Here since we already read a command at the start of the function (which is normal/medium priority command) the next command that can be if it all if it is received is super high priority command i.e. SCH_SCHEDULE_RESTART_CYCLE. In fact these changes were there earlier I just changed the prototype of the function get_sched_cmd_noblk.